flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/BigQuerySink.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.sink;
+
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.connector.base.DeliveryGuarantee;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Class wrapping BigQuery sinks with appropriate configurations. */
+public class BigQuerySink {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BigQuerySink.class);
+
+    public static Sink get(BigQuerySinkConfig sinkConfig, StreamExecutionEnvironment env) {
+        if (sinkConfig.getDeliveryGuarantee() == DeliveryGuarantee.EXACTLY_ONCE) {
```

Comment:
what if delivery guarantee is NONE? https://nightlies.apache.org/flink/flink-docs-master/api/java/org/apache/flink/connector/base/DeliveryGuarantee.html#NONE

Reply:
The default sink will offer either at-least-once or at-most-once depending on the checkpointing mode. Eventual data consistency in the destination (like BQ table) is governed by the combination of sink delivery guarantee and checkpointing mode.

1. If checkpointing is disabled, then regardless of delivery guarantee the sink will offer at-most-once consistency.
2. For default sink to offer at-least-once, delivery guarantee must be at-least-once and checkpointing mode should be at-least-once or exactly-once.
3. Delivery guarantee must be exactly-once isnt supported yet.

Reply:
yeah that makes sense to me but the user can pass a deliveryGuarentee.NONE during configuration. i'm wodnering if we want to throw an error there or if we want to just treat it as ALO

Reply:
Good point. Let's restrict to at-least-once for now, and we can extend this sink to support deliveryGuarentee.NONE in future.
Just to summarize though, theoretically this same sink can offer both at-least-once and at-most-once based on checkpointing mode.